File: app/src/main/java/com/example/marvelworld/ui/MarvelListViewModelFactory.java
package com.example.marvelworld.ui;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MarvelListViewModelFactory implements ViewModelProvider.Factory {

    private String apiKey;

    public MarvelListViewModelFactory(String apiKey) {
        this.apiKey = apiKey;
    }

    @NonNull @Override public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(MarvelListViewModel.class)) {
            return (T) new MarvelListViewModel(apiKey);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/example/marvelworld/ui/MarvelListViewModelFactory.kt
package com.example.marvelworld.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MarvelListViewModelFactory(private val apiKey: String) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(MarvelListViewModel::class.java)) {
            return MarvelListViewModel(apiKey) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}